src/main/java/httpserver/HttpRequestParser.java

```diff
@@ -0,0 +1,21 @@
+package httpserver;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class HttpRequestParser {
+    public static HttpRequest fromString(String rawRequest){
+        List<String> chunkedRawRequest = Arrays.asList(rawRequest.split("\r\n"));
+        String[] statusLine = chunkedRawRequest.get(0).split(" ");
+        String body = "";
+
+        if(chunkedRawRequest.contains("")){
+            body = chunkedRawRequest.get(chunkedRawRequest.size() - 1);
```

Comment:
Why `- 1`

Reply:
If a body is present, it will always be the last string in a valid http request.

Reply:
I see. Is there some way that we could elucidate why we are getting the last element then?

I.e., how could this be refactored to be more declarative instead of procedural?